code/game/objects/items/tools/weldingtool.dm

```diff
@@ -194,10 +196,20 @@
 		return FALSE
 
 
+//Toggles the welding value.
+/obj/item/weldingtool/proc/set_welding(new_value)
+	if(welding == new_value)
+		return
+	. = welding
+	welding = new_value
+	lighting_overlay_toggle_on(welding)
+
+
 //Turns off the welder if there is no more fuel (does this really need to be its own proc?)
 /obj/item/weldingtool/proc/check_fuel(mob/user)
 	if(get_fuel() <= 0 && welding)
-		set_light(0)
+		var/datum/component/overlay_lighting/overlay_lighting = GetComponent(/datum/component/overlay_lighting)
+		overlay_lighting.turn_off()
```

Comment:
why not just `lighting_overlay_toggle_on(FALSE)`?

Reply:
The component in Skyrat used `GetComponent` instead of signals. This was one occurrence I missed, but it's fixed now, thanks for pointing it out.